homeassistant/helpers/area_registry.py

```diff
@@ -197,8 +197,19 @@ def async_get_area(self, area_id: str) -> AreaEntry | None:
 
     @callback
     def async_get_area_by_name(self, name: str) -> AreaEntry | None:
```

Comment:
This should be added as a a new function so we don't have unintended side effects.

We should then update template method to call it. Probably also update suggested_area processing in device registry to use it.